lib/Fhp/FinTs.php

```diff
@@ -37,376 +37,409 @@
+	public function setTimeouts($connect, $response)
+	{
 		$this->timeoutConnect = $connect;
 		$this->timeoutResponse = $response;
 	}
 
-    /**
-     * Gets array of all accounts.
-     *
-     * @return Model\Account[]
-     */
-    public function getAccounts() {
-        $dialog = $this->getDialog(false);
-        $result = $dialog->syncDialog();
-        $this->end();
-        $this->bankName = $dialog->getBankName();
-        $accounts = new GetAccounts($result);
-
-        return $accounts->getAccountsArray();
-    }
-
-    /**
-     * Gets array of all SEPA Accounts.
-     *
-     * @return Model\SEPAAccount[]
-     * @throws \CurlException
-     */
-    public function getSEPAAccounts() {
-        $dialog = $this->getDialog(false);#, $this->tanMechanism);
+	/**
+	 * Gets array of all accounts.
+	 *
+	 * @return Model\Account[]
+	 */
+	public function getAccounts()
+	{
+		$dialog = $this->getDialog(false);
+		$result = $dialog->syncDialog();
+		$this->end();
+		$this->bankName = $dialog->getBankName();
+		$accounts = new GetAccounts($result);
+
+		return $accounts->getAccountsArray();
+	}
+
+	/**
+	 * Gets array of all SEPA Accounts.
+	 *
+	 * @return Model\SEPAAccount[]
+	 * @throws \CurlException
+	 */
+	public function getSEPAAccounts()
+	{
+		$dialog = $this->getDialog(false);#, $this->tanMechanism);
 		#$dialog->endDialog(); //probably not required
-		$dialog->syncDialog(true, $this->tanMechanism);
+		$dialog->syncDialog($this->tanMechanism, $this->tanMediaName);
 		$dialog->endDialog();
-        $dialog->initDialog();
+		$dialog->initDialog($this->tanMediaName);
 
-        $message = $this->getNewMessage(
-            $dialog,
-            array(new HKSPA(3)),
-            array(AbstractMessage::OPT_PINTAN_MECH => $this->getUsedPinTanMechanism($dialog))
-        );
+		$message = $this->getNewMessage(
+			$dialog,
+			array(new HKSPA(3)),
+			array(AbstractMessage::OPT_PINTAN_MECH => $this->getUsedPinTanMechanism($dialog))
+		);
 
 		$this->logger->info('');
 		$this->logger->info('HKSPA (SEPA accounts) initialize');
-        $result = $dialog->sendMessage($message);
+		$result = $dialog->sendMessage($message);
 		$this->logger->info('HKSPA end');
 
-        $sepaAccounts = new GetSEPAAccounts($result->rawResponse);
+		$sepaAccounts = new GetSEPAAccounts($result->rawResponse);
 
-        return $sepaAccounts->getSEPAAccountsArray();
-    }
+		return $sepaAccounts->getSEPAAccountsArray();
+	}
 
-	public function getVariables(){
-        $dialog = $this->getDialog(false);
-		$result = $dialog->syncDialog(false);
-		$this->end();
+	public function getVariables()
+	{
+		$dialog = $this->getDialog(false);
+		$response = $dialog->syncDialog();
+		// $this->end();
 
-		$R = new GetVariables($result->rawResponse);
-		return $R->get();
+		$vars = new GetVariables($response->rawResponse);
+		return $vars->get();
 	}
 
-	public function end(){
-		if(!$this->dialog)
+	public function getTANRequest()
```

Comment:
This may be useful in stateless applications. I did not need it yet though, was merely leftover from testing.

Reply:
:heavy_check_mark: